src/main/java/com/olgaboiar/mint/Header.java

```diff
@@ -30,25 +31,58 @@ String getDate() {
         return date;
     }
 
-    String prepareHeaders() {
-        if (allowedMethods != null) {
-            String methods = String.join(",", allowedMethods);
-            String accessControl = "Allow: " + methods;
-            headersToSend = statusLine + "\n"
-                    + contentType + "\n"
-                    + date + "\n"
-                    + accessControl + "\n"
-                    + blankLine;
-        } else {
-            headersToSend = statusLine + "\n"
-                + contentType + "\n"
-                + date + "\n"
-                + blankLine;
+    public ArrayList<String> getHeaders() {
+        ArrayList<String> headers = new ArrayList<String>();
+        headers.add(getStatusLine());
+        headers.add(getContentType());
+        headers.add(getDate());
+        if (allowedMethodsExist()) {
+            String aloowHeader = createAllowHeader();
+            headers.add(aloowHeader);
+        }
+        if (redirectExist()) {
+            String redirectHeader = createRedirectHeader();
+            headers.add(redirectHeader);
         }
+        return headers;
+    }
+
+    public String prepareHeaders() {
+        String headersToSend = String.join("\n", getHeaders());
         return headersToSend;
     }
 
-    public void setAllowMethods(String[] allowedMethods1) {
-        allowedMethods = allowedMethods1;
+    public boolean allowedMethodsExist () {
+        if (allowedMethods != null) {
+            return true;
+        }
+        return false;
+    }
+
+    public String createAllowHeader() {
+        String allMethods = String.join(",", allowedMethods);
+        String allowHeader = "Allow: " + allMethods;
+        return allowHeader;
+
+    }
+
+    public String createRedirectHeader () {
+        String redirectHeader = "Location: " + redirect;
+        return redirectHeader;
+    }
+
+    public boolean redirectExist () {
+        if (redirect != null) {
+            return true;
+        }
+        return false;
```

Comment:
My personal preference when I see this pattern: `if (clause) return true else return false`, I like to simplify it to `return (clause)`.

Reply:
refactored in https://github.com/olgaboiar/MintHTTPServer/pull/10/commits/9bf42609c891d15573859aedbc3d47bd82db5139